src/main/java/com/prgrms/vouchermanager/VoucherManagerApplication.java

```diff
@@ -0,0 +1,30 @@
+package com.prgrms.vouchermanager;
+
+import com.prgrms.vouchermanager.controller.VoucherController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.ansi.AnsiOutput;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class VoucherManagerApplication implements CommandLineRunner {
+
+    private final VoucherController controller;
+
+    public VoucherManagerApplication(VoucherController controller) {
+        this.controller = controller;
+    }
+
+    @Autowired
```

Comment:
```suggestion
    @Autowired
    public VoucherManagerApplication(VoucherController controller) {
        this.controller = controller;
    }
```

생성자 주입을 위해서 @Autowired를 사용하신거겠죠? 아래로 내려와 있어서 여쭤봅니다.!

Reply:
맞습니다! 제가 실수를 발견하지 못했네요 ㅠㅠ 그런데 어떻게 제대로 주입돼서 동작한건지 궁금합니다!

Reply:
> 맞습니다! 제가 실수를 발견하지 못했네요 ㅠㅠ 그런데 어떻게 제대로 주입돼서 동작한건지 궁금합니다!

제가 아는 바로는 @Autowired 를 명시적으로 작성하지 않아도 생성자를 통해 VoucherController를 파라미터로 갖는 경우, 
자동적으로 @Autowired의 효과로 @Controller를 마킹하여 Bean으로 등록한 VoucherController를 주입 받을 수 있습니다.
현재 코드는 DI하는 방식은 세 가지 방식 중 필드 주입, 생성자 주입, Setter주입 중에 생성자 주입의 형태로 볼 수 있습니다.

이러한 과정이 이루어 질 수 있는 이유는 @SpringBootApplication 이 있기 때문입니다.

한번 천천히 뜯어보시는 것을 추천드립니다. 위 어노테이션에 어떤 어노테이션을 포함하고 있는지, 각 어노테이션들이 어떠한 역할을 하고 있는지 등등 !